returnn/tf/layers/basic.py

```diff
@@ -2945,6 +2945,50 @@ def get_out_data_from_opts(cls, name, axis, repetitions, sources=(), **kwargs):
     return data
 
 
+class TileLayer(_ConcatInputLayer):
+  """
+  A wrapper around tf.tile
+  """
+  layer_class = "tile"
+
+  def __init__(self, multiples, **kwargs):
+    """
+    :param dict[str, int] multiples: number of multiples per axis (axis provided as str)
+    """
+    from returnn.tf.util.data import DimensionTag
+    super(TileLayer, self).__init__(**kwargs)
+    self.multiples = multiples
+    input_data = self.input_data
+
+    multiples_full = [1] * input_data.batch_ndim
+    for axis, multiple in multiples.items():
+      a = input_data.get_axis_from_description(axis, allow_int=False)
+      if multiple != 1:
+        assert a not in input_data.get_dynamic_axes(), "Tiling of dynamic axes not yet implemented"
+        assert a != input_data.batch_dim_axis, "Tiling of batch axis not yet implemented"
+      multiples_full[a] *= multiple
+
+    self.output.placeholder = tf.tile(input_data.placeholder, tf.constant(multiples_full))
```

Comment:
Why the explicit `constant`? Keep this simple.
```suggestion
    self.output.placeholder = tf.tile(input_data.placeholder, multiples_full)
```